### DISCUSSION TITLE
Question of the variability between coffee-editor and modelserver

### DISCUSSION THREAD

**Discussion by kun7758** [eclipse-emfcloud/emfcloud#144] [Coffee Editor]
hello,I hava a question about there is large difference between modelserver 's (module of project coffee-editor) master branch and the master branch of project modelserver.

I think the version of modelserver( module of project coffee-editor) is newer than the project modelserver.

for example, the Class:  org.eclipse.emfcloud.modelserver.client.ModelServerClient

would you give me a response as soon as possible, appreciate

here is the code in coffee-editor
![111](https://user-images.githubusercontent.com/101698445/158540649-519e494d-3fbd-4873-8237-051c7eb8e403.png)

here is the code in modelserver
![222](https://user-images.githubusercontent.com/101698445/158540683-6e1af168-c72a-460c-9de2-b40932a34c77.png)

**Comment by eneufeld** [comment]
Hi @kunxiao1415 ,
The coffee editor is using the modelserver as a dependency.
As you can see the dependency used by the coffee editor is currently 0.7.0.202201070845 .
Obviously it is not possible to update the modelserver dependency as soon as there is a new snapshot version of the modelserver.
The modelserver has introduced a new API which should solve some issue we have in the coffee editor tree editor use.
Any contribution is always welcome.

**Reply by kun7758** [reply]
Now that I have modified the source code of the modelserver project, how should I apply it to the coffe-editor?

**Reply by eneufeld** [reply]
Hi @kunxiao1415 ,
could you provide more context.
What did you modify? What do you want to apply to the coffee-editor?
In general the coffee editor is adapting the modelserver to work with its model, so this is where you should do adaptations. If you did modify the modelserver core then you need to build it as part of the coffee editor build, thus adapt the parent pom to include your modified modelserver projects and remove the modelserver dependencies that are referencing the original project.
